packages/aws-cdk/test/api/state-machine-hotswap-deployments.test.ts

```diff
@@ -0,0 +1,226 @@
+  // GIVEN
+  currentCfnStack.setTemplate({
+    Resources: {
+      Machine: {
+        Type: 'AWS::StepFunctions::StateMachine',
+        Properties: {
+          DefinitionString: {
+            'Fn::Join': [
+              '',
+              [
+                '{ "Prop" : "old-value" }',
+              ],
+            ],
+          },
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Machine: {
+          Type: 'AWS::StepFunctions::StateMachine',
+          Properties: {
+            DefinitionString: {
+              'Fn::Join': [
+                '',
+                [
+                  '{ "Prop" : "new-value" }',
+                ],
+              ],
+            },
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await tryHotswapDeployment(mockSdkProvider, {}, currentCfnStack, cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).not.toBeUndefined();
+  expect(mockUpdateMachineCode).toHaveBeenCalledWith({
+    definition: '{"Prop":"new-value"}',
+    stateMachineArn: 'mock-machine-resource-id', // the sdk will convert the ID to the arn in a production environment
+  });
+});
+
+test('does not call the updateStateMachine() API when it receives a change to a parameter that is not the definitionString in a state machine', async () => {
+  // GIVEN
+  currentCfnStack.setTemplate({
+    Resources: {
+      Machine: {
+        Type: 'AWS::StepFunctions::StateMachine',
+        Properties: {
+          DefinitionString: {
+            'Fn::Join': [
+              '',
+              [
+                '{ "Prop" : "old-value" }',
+              ],
+            ],
+          },
+          LoggingConfiguration: { // non-definitionString property
+            IncludeExecutionData: true,
+          },
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Machine: {
+          Type: 'AWS::StepFunctions::StateMachine',
+          Properties: {
+            DefinitionString: {
+              'Fn::Join': [
+                '',
+                [
+                  '{ "Prop" : "old-value" }',
+                ],
+              ],
+            },
+            LoggingConfiguration: {
+              IncludeExecutionData: false,
+            },
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await tryHotswapDeployment(mockSdkProvider, {}, currentCfnStack, cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).toBeUndefined();
+  expect(mockUpdateMachineCode).not.toHaveBeenCalled();
+});
+
+function cdkStackArtifactOf(testStackArtifact: Partial<TestStackArtifact> = {}): cxapi.CloudFormationStackArtifact {
+  return testStack({
+    stackName: STACK_NAME,
+    ...testStackArtifact,
+  });
+}
```

Comment:
There's a _lot_ of duplication in these 3 test files, starting from the local variables list, the initial `beforeEach()`, and ending with this `cdkStackArtifactOf()` function.

Any chance you can work on reducing this duplication somehow?